packages/resilience/src/retry.ts

```diff
@@ -0,0 +1,35 @@
+import * as asyncRetry from 'async-retry';
+import { Context, Middleware, MiddlewareIn, MiddlewareOut } from './middleware';
+
+const defaultRetryCount = 3;
+
+/**
+ * Helper method to build a retry middleware.
+ * @param retryCount - Number of retry attempts. Default value is 3.
+ * @returns The middleware adding a retry to the function.
+ */
+export function retry<ReturnType, ContextType extends Context>(
+  retryCount: number = defaultRetryCount
+): Middleware<ReturnType, ContextType> {
+  return function (
+    options: MiddlewareIn<ReturnType, ContextType>
+  ): MiddlewareOut<ReturnType> {
+    return () =>
+      asyncRetry.default(
+        async bail => {
+          try {
+            return await options.fn();
+          } catch (error) {
+            // Don't retry on error statuses where a second attempt won't help
+            const status = error.response.status;
```

Comment:
[req] make the acces on `error.response.status` optional with `?` - you may add a low level log statement informing that the property is not present.